### PR TITLE
Add test_bigquery suite

### PR DESCRIPTION
**Pull request by MattDelac**
Signed-off-by: Matt Delacour <matt.delacour@shopify.com>

**What this PR does / why we need it**:
Add tests for the various retrieval method of BigQueryRetrievalJob

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/tests/infra/offline_stores/test_bigquery.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
+import random
```
[the remark this answers is not in the document]
Comment (MattDelac) [reply sdk/python/tests/infra/offline_stores/test_bigquery.py:1]: I don't think I understand. This is under `/tests/infra/offline_stores/`

Is there another test folder ?

[the remark this answers is not in the document]
Comment (tsotnet) [reply sdk/python/tests/infra/offline_stores/test_bigquery.py:1]: `sdk/python/tests` does not have `infra` folder. Unit tests go to `unit`, while integration tests go to `integration` sub-directory. The integration tests are further split into different stages, one of which is offline_store tests. Thus Felix's suggestion is to move this test under `sdk/python/tests/integration/offline_store`.

File: sdk/python/tests/infra/offline_stores/test_bigquery.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    store.apply([driver, driver_fv])
+    return store.get_historical_features(
+        entity_df=entity_df,
+        features=["driver_stats:avg_daily_trips"],
+        full_feature_names=False,
+    )
+
+
+@pytest.mark.integration
+def test_bigquery_to_df(environment, driver_input_df, entity_df, expected_df):
```
Comment (tsotnet) [inline sdk/python/tests/infra/offline_stores/test_bigquery.py:107]: We already test to_df and to_arrow in `test_universal_historical_retrieval.py` for BigQuery (as well as all other supported offline stores). Looks like `to_bigquery` is the only new function being tested here, so we can probably remove `test_bigquery_to_df` and `test_bigquery_to_arrow`.

File: sdk/python/tests/infra/offline_stores/test_bigquery.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        expected_df.sort_values(by=["driver_id"]).reset_index(drop=True),
+        actual_df[expected_df.columns]
+        .sort_values(by=["driver_id"])
+        .reset_index(drop=True),
+        check_dtype=False,
+    )
+
+
+@pytest.mark.integration
+def test_bigquery_to_bigquery(environment, driver_input_df, entity_df, expected_df):
```
Comment (tsotnet) [inline sdk/python/tests/infra/offline_stores/test_bigquery.py:137]: @achals is this the best way to test `to_bigquery`? Or is there a suggested way of extending universal feature repo tests?

**Comment by adchia** [comment]
Were we blocked on something here?

**Comment by MattDelac** [comment]
> Were we blocked on something here?

I don't think so. Thought that it was covered by the universal test suite and never got back to it.

My bandwidth is pretty low for the next few weeks. Do you think it's a "must have" or a "nice to have" ?

**Comment by adchia** [comment]
def a nice to have. Wouldn't prioritize it. Just wanted to check in on it.

Don't think the universal test suite specifically tests the `to_bigquery` flow, but otherwise should be covered!

**Comment by achals** [comment]
Hi @MattDelac, are you still planning on working on this PR? Should we keep it open?

**Pull request closed without merging**